helper.go

```diff
@@ -107,6 +120,77 @@ func InitHelper(config *Config) (*Helper, error) {
 	}, nil
 }
 
+// symlinkAuxiliaryProviders discovers auxiliary provider binaries, used in
+// multi-provider tests, and symlinks them to the plugin directory.
+//
+// Auxiliary provider binaries should be included in the provider source code
+// directory, under the path terraform.d/plugins/$GOOS_$GOARCH/provider-name.
+//
+// The environment variable TF_ACC_PROVIDER_ROOT_DIR must be set to the path of
+// the provider source code directory root in order to use this feature.
+func symlinkAuxiliaryProviders(pluginDir string) error {
+	providerRootDir := os.Getenv("TF_ACC_PROVIDER_ROOT_DIR")
+	if providerRootDir == "" {
+		// common case; assume intentional and do not log
+		return nil
+	}
+
+	_, err := os.Stat(filepath.Join(providerRootDir, "terraform.d", "plugins"))
+	if err != nil {
```

Comment:
ensure not exist with `if os.IsNotExist(err) {`

Reply:
to be robust throw in an `else if err !=nil { fmt.Errorf("Unexpected error: %s", err) }`